src/plugins/charts/public/static/components/color_picker.tsx

```diff
@@ -74,54 +74,87 @@ export const legendColors: string[] = [
-        id="charts.colorPicker.setColor.screenReaderDescription"
-        defaultMessage="Set color for value {legendDataLabel}"
-        values={{ legendDataLabel: label }}
-      />
-    </span>
-    <div className="visColorPicker__value" role="listbox">
-      {legendColors.map((color) => (
-        <EuiIcon
-          role="option"
-          tabIndex={0}
-          type="dot"
-          size="l"
-          color={selectedColor}
-          key={color}
-          aria-label={color}
-          aria-describedby={`${id}ColorPickerDesc`}
-          aria-selected={color === selectedColor}
-          onClick={(e) => onChange(color, e)}
-          onKeyPress={(e) => onChange(color, e)}
-          className={classNames('visColorPicker__valueDot', {
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            'visColorPicker__valueDot-isSelected': color === selectedColor,
-          })}
-          style={{ color }}
-          data-test-subj={`visColorPickerColor-${color}`}
+const euiColors = euiPaletteColorBlind({ rotations: 4, order: 'group' });
+
+export const ColorPicker = ({
+  onChange,
+  color: selectedColor,
+  id,
+  label,
+  useLegacyColors = true,
+  colorIsOverwritten = true,
+}: ColorPickerProps) => {
+  const legendColors = useLegacyColors ? legacyColors : euiColors;
+
+  return (
+    <div className="visColorPicker">
+      <span id={`${id}ColorPickerDesc`} className="euiScreenReaderOnly">
+        <FormattedMessage
+          id="charts.colorPicker.setColor.screenReaderDescription"
+          defaultMessage="Set color for value {legendDataLabel}"
+          values={{ legendDataLabel: label }}
         />
-      ))}
+      </span>
+      <div className="visColorPicker__value" role="listbox">
+        {legendColors.map((color) => (
+          <EuiIcon
+            role="option"
+            tabIndex={0}
+            type="dot"
+            size="l"
+            color={selectedColor}
+            key={color}
+            aria-label={color}
+            aria-describedby={`${id}ColorPickerDesc`}
+            aria-selected={color === selectedColor}
+            onClick={(e) => onChange(color, e)}
+            onKeyPress={(e) => onChange(color, e)}
+            className={classNames('visColorPicker__valueDot', {
+              // eslint-disable-next-line @typescript-eslint/naming-convention
+              'visColorPicker__valueDot-isSelected': color === selectedColor,
+            })}
+            style={{ color }}
+            data-test-subj={`visColorPickerColor-${color}`}
+          />
+        ))}
+      </div>
+      {legendColors.some((c) => c === selectedColor) && colorIsOverwritten && (
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            size="s"
+            onClick={(e: any) => onChange(null, e)}
+            onKeyPress={(e: any) => onChange(null, e)}
+          >
+            <FormattedMessage id="charts.colorPicker.clearColor" defaultMessage="Clear color" />
```

Comment:
Instead of using "clear" should we use something like "reset"? "Clear", I think, implies that we're removing the color completely but it seems like what we're doing is returning it back to the original color that we set. 
```suggestion
            <FormattedMessage id="charts.colorPicker.clearColor" defaultMessage="Reset color" />
```